linkup/src/versioning.rs

```diff
@@ -8,48 +8,86 @@ pub enum VersionError {
 
 #[derive(Debug, Clone)]
 pub struct Version {
-    major: u16,
-    minor: u16,
-    patch: u16,
+    pub major: u16,
+    pub minor: u16,
+    pub patch: u16,
+    pub pre_release: Option<String>,
+}
+
+impl Version {
+    pub fn is_beta(&self) -> bool {
+        if let Some(pre_release) = &self.pre_release {
+            return pre_release.starts_with("next-");
+        }
+
+        false
+    }
 }
 
 impl PartialEq for Version {
     fn eq(&self, other: &Self) -> bool {
-        self.major == other.major && self.minor == other.minor && self.patch == other.patch
+        self.major == other.major
+            && self.minor == other.minor
+            && self.patch == other.patch
+            && self.pre_release == other.pre_release
     }
 }
 
 impl PartialOrd for Version {
     fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
-        match (
-            self.major.cmp(&other.major),
-            self.minor.cmp(&other.minor),
-            self.patch.cmp(&other.patch),
-        ) {
-            (std::cmp::Ordering::Equal, std::cmp::Ordering::Equal, ord) => Some(ord),
-            (std::cmp::Ordering::Equal, ord, _) => Some(ord),
-            (ord, _, _) => Some(ord),
+        match (&self.pre_release, &other.pre_release) {
+            (Some(a), Some(b)) => a.cmp(b).into(),
+            (Some(_), None) => {
+                // pre-release is always lower than stable
+                Some(std::cmp::Ordering::Less)
+            }
+            (None, Some(_)) => {
+                // stable is always higher than pre-release
+                Some(std::cmp::Ordering::Greater)
+            }
```

Comment:
This is more to protect against any accidental updates from beta channel into stable, and the other way around.
It shouldn't happen, since the `available_update` fetches from the same channel. But might be good to have.